server/msteams/client_test.go

```diff
@@ -0,0 +1,161 @@
+							User: &msgraph.Identity{
+								ID: &teamsUserID,
+							},
+						},
+					},
+				},
+			},
+			ExpectedResult: Message{
+				UserID:          teamsUserID,
+				UserDisplayName: teamsUserDisplayName,
+				ReplyToID:       teamsReplyID,
+				Text:            content,
+				Subject:         subject,
+				LastUpdateAt:    time.Time{},
+				Attachments: []Attachment{
+					{
+						ContentType: attachmentContentType,
+						Content:     attachmentContent,
+						Name:        attachmentName,
+						ContentURL:  attachmentURL,
+					},
+				},
+				Reactions: []Reaction{
+					{
+						Reaction: reactionType,
+						UserID:   teamsUserID,
+					},
+				},
+				ChannelID: testutils.GetChannelID(),
+				TeamID:    "mockTeamsTeamID",
+				ChatID:    "mockChatID",
+			},
+		},
+		{
+			Name: "ConvertToMessage: With no data filled",
+			ChatMessage: &msgraph.ChatMessage{
+				LastModifiedDateTime: &time.Time{},
+			},
+			ExpectedResult: Message{
+				Attachments:  []Attachment{},
+				Reactions:    []Reaction{},
+				LastUpdateAt: time.Time{},
+				ChannelID:    testutils.GetChannelID(),
+				TeamID:       "mockTeamsTeamID",
+				ChatID:       "mockChatID",
+			},
+		},
+	} {
+		t.Run(test.Name, func(t *testing.T) {
+			assert := assert.New(t)
+			resp := convertToMessage(test.ChatMessage, "mockTeamsTeamID", testutils.GetChannelID(), "mockChatID")
+
+			assert.Equal(test.ExpectedResult, *resp)
+		})
+	}
+}
+
+func TestGetResourceIds(t *testing.T) {
+	for _, test := range []struct {
+		Name           string
+		Resource       string
+		ExpectedResult ActivityIds
+	}{
+		{
+			Name:     "GetResourceIds: With prefix chats(",
+			Resource: "chats('19:8ea0e38b-efb3-4757-924a-5f94061cf8c2_97f62344-57dc-409c-88ad-c4af14158ff5@unq.gbl.spaces')/messages('1612289765949')",
+			ExpectedResult: ActivityIds{
+				ChatID:    "19:8ea0e38b-efb3-4757-924a-5f94061cf8c2_97f62344-57dc-409c-88ad-c4af14158ff5@unq.gbl.spaces",
+				MessageID: "1612289765949",
+			},
+		},
+		{
+			Name:     "GetResourceIds: Without prefix chats(",
+			Resource: "teams('fbe2bf47-16c8-47cf-b4a5-4b9b187c508b')/channels('19:4a95f7d8db4c4e7fae857bcebe0623e6@thread.tacv2')/messages('1612293113399')/replies('19:zOtXfpDMWANo7-9CHuzHdM7WPSamQejH0Vydj0U-Yho1')",
+			ExpectedResult: ActivityIds{
+				TeamID:    "fbe2bf47-16c8-47cf-b4a5-4b9b187c508b",
+				ChannelID: "19:4a95f7d8db4c4e7fae857bcebe0623e6@thread.tacv2",
+				MessageID: "1612293113399",
+				ReplyID:   "19:zOtXfpDMWANo7-9CHuzHdM7WPSamQejH0Vydj0U-Yho1",
+			},
+		},
+		{
+			Name:     "GetResourceIds: Resource with multiple '/'",
+			Resource: "/////19:4a95f7d8db4c4e7fae857bcebe0623e6@thread.tacv2///",
+		},
+		{
+			Name: "GetResourceIds: Empty resource",
+		},
+		{
+			Name:     "GetResourceIds: Resource with small length",
+			Resource: "ID",
+		},
+		{
+			Name:     "GetResourceIds: Resource with large length",
+			Resource: "very-long-teams-ID-with-very-long-chat-ID",
+			ExpectedResult: ActivityIds{
```

Comment:
I think this expected result makes no sense, I would expect that this doesn't work, probably returning an empty result, like other cases. I prefer a failing tests that makes sense, than a not failing test that follows the current wrong behavior.

Reply:
@jespino  should I change the function implementation, as the current implementation will not return an empty result or should we remove this test case?

Reply:
Yes, that is the idea. If the behavior is not correct, we should fix the behavior, not change the test. I'm ok if you make the changes for the test to fail and I can jump in later and fix the implementation. However you prefer. But I don't want "test that verify the current behavior", I want "test that verify the correct behavior".